projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -5842,9 +5842,18 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     /**
      * @hidden @internal
      */
-    public navigateToAddedRow(event) {
-        this.navigateTo(this.dataView.length - 1);
-        this.addRowSnackbar.hide();
+    public showSnackbarFor(id: number) {
+        if (id === -1) {
+            this.addRowSnackbar.actionText = '';
+            this.addRowSnackbar.show();
+            return;
+        }
+        this.addRowSnackbar.actionText = this.snackbarActionText;
+        this.addRowSnackbar.show();
+        this.addRowSnackbar.onAction.subscribe(() => {
```

Comment:
This will subscribe a new listener every time a new row is added so the event will fire multiple times when clicked after a few rows are added. Instead you can subscribe once, cache the last add row index in a private prop and use it  to navigate to the correct index.